File: problems/algorithms/function-factorial/php/response/spec/factorial.spec.php
<?php

require __DIR__ . '/../src/factorial.php';

describe('Number Tools', function () {
  it('calculating factorial of 1', function () {
    expect(factorial(1))->toBe(1);
  });

  it('calculating factorial of 2', function () {
    expect(factorial(2))->toBe(2);
  });

  it('calculating factorial of 3', function () {
    expect(factorial(3))->toBe(6);
  });

  it('calculating factorial of 4', function () {
    expect(factorial(4))->toBe(24);
  });
});

File: problems/algorithms/function-factorial/ruby/response/spec/factorial.spec.rb
require_relative "../src/factorial"

  describe ".factorial" do
    context "calc factorial of 1" do
      it "return 1" do
        expect(factorial(1)).to eql(1)
      end
    end
  
    context "calc factorial of 2" do
      it "return 2" do
        expect(factorial(2)).to eql(2)
      end
    end

    context "calc factorial of 3" do
      it "return 6" do
        expect(factorial(3)).to eql(6)
      end
    end

    context "calc factorial of 4" do
      it "return 24" do
        expect(factorial(4)).to eql(24)
      end
    end
  end
 